File: src/parser/helpers/rollback.js
"use strict";

/**
 * Rollback loop index/column to re-run parser at same iteration.
 *
 * @param  {object} S - State object.
 * @return {undefined} - Nothing is returned.
 */
module.exports = S => {
	S.i--;
	S.column--;
};

File: src/parser/nim/parser/helpers/rollback.nim
from types import State

# Rollback loop index/column to re-run parser at same iteration.
#
# @param  {object} S - State object.
# @return - Nothing is returned.
proc rollback*(S: State) =
    S.i = S.i - 1
    S.column = S.column - 1
